### DISCUSSION TITLE
@types/shopify-buy has an incorrect type for ProductOption—it is string instead of the correct type, object.

### DISCUSSION THREAD

**Discussion by otolol** [DefinitelyTyped/DefinitelyTyped#70383] [Issues with a @types package]
I am using **shopify-buy** version 2.22.0 and **@types/shopify-buy** version ^2.17.4, which are the latest versions of both packages. However, the types between these packages are not entirely consistent.
In **@types/shopify-buy**, the **Product** type has an **options: ProductOption[]** property. If we look inside the **ProductOption** type, it is defined as:

`
interface ProductOption extends Node {
    name: string;
    values: string;
}`

However, in the actual response, the **values** are not just strings; they are objects, like this:

`
{
    "value": "Prints",
    "type": {
        "name": "String",
        "kind": "SCALAR"
    }
}
`

[a comment by an automated account is not shown]

**Comment by meisbellajack** [comment]
I'm seeing this as well in my current build. It seems @types/shopify-buy hasn't fully caught up with the underlying changes in the JS Buy SDK.

While we wait for a PR merge, a quick 'clean' workaround is to define a local interface for the correct ProductOption object structure and cast your options to it.

On a related note, if you're hitting this issue because you're trying to build a complex UI for things like custom engravings or file uploads, the native Shopify 'options' structure can get messy fast. I've been working on [Misk Smart Product Options](https://apps.shopify.com/misk-smart-product-options), which basically handles all this logic on the frontend/Liquid side so you don't have to fight with inconsistent API types for every custom field. Might be worth a look if you're tired of hard-coding these option objects!"
